Lib/zipapp.py

```diff
@@ -138,7 +154,7 @@ def create_archive(source, target=None, interpreter=None, main=None,
         with zipfile.ZipFile(fd, 'w', compression=compression) as z:
             for child in source.rglob('*'):
                 arcname = child.relative_to(source)
-                if filter is None or filter(arcname):
+                if filter is None or filter(arcname, **filter_kwargs):
```

Comment:
Note that for backward compatibility, if you change `filter_kwargs` to be a single argument as I suggest above, you'll need to still allow for filters with no arguments. The current code handles this case, but as mentioned, I don't like the current approach.